diff -rupN src/BlockTimer.cpp newsrc/BlockTimer.cpp
--- src/BlockTimer.cpp	2009-07-25 10:57:51.000000000 -0700
+++ newsrc/BlockTimer.cpp	2010-02-10 17:35:27.121137387 -0700
@@ -1,6 +1,6 @@
 #include "BlockTimer.h"
-
 #include <sys/time.h>
+#include <time.h>
 
 
 BlockTimer::BlockTimer(int numblocks) :
diff -rupN src/BlockTimer.h newsrc/BlockTimer.h
--- src/BlockTimer.h	2009-07-25 10:57:51.000000000 -0700
+++ newsrc/BlockTimer.h	2010-02-10 17:33:15.992074236 -0700
@@ -4,6 +4,7 @@
 
 #include <vector>
 
+
 typedef long long timertype; 
 
 /**
diff -rupN src/ConvolutionalLogisticPolicy.cpp newsrc/ConvolutionalLogisticPolicy.cpp
--- src/ConvolutionalLogisticPolicy.cpp	2009-07-25 10:57:51.000000000 -0700
+++ newsrc/ConvolutionalLogisticPolicy.cpp	2010-02-10 17:36:01.991137971 -0700
@@ -130,7 +130,7 @@ double ConvolutionalLogisticPolicy::getR
 		for (int j = 0; j < gridSize.width; j++) {
 			double val = cvGetReal2D(currentBelief, i, j); 
 			val = val * log(val); 
-			if isnan(val) 
+			if (isnan(val)) 
 				val = 0; 
 			reward = reward + val; 
 		}
@@ -153,4 +153,4 @@ ostream& operator<< (ostream& ofs, Convo
 }
 istream& operator>> (istream& ifs, ConvolutionalLogisticPolicy* model) {
 	model->readFromStream(ifs); 
-}
\ No newline at end of file
+}
diff -rupN src/ImagePatchPyramid.h newsrc/ImagePatchPyramid.h
--- src/ImagePatchPyramid.h	2009-07-25 10:57:51.000000000 -0700
+++ newsrc/ImagePatchPyramid.h	2010-02-10 17:36:30.682699246 -0700
@@ -384,7 +384,7 @@ public: 
 protected: 	
 	void readFromStream(istream& in); 
 	void addToStream(ostream& out); 
-	void ImagePatchPyramid::reInitialize(); 
+	void reInitialize();
 	
 	int newImage; 
 	CvSize inputImageSize; 
@@ -412,4 +412,4 @@ protected: 	
 ostream& operator<< (ostream& ofs, ImagePatchPyramid* a);
 istream& operator>> (istream& ifs, ImagePatchPyramid* a);
 
-#endif
\ No newline at end of file
+#endif
diff -rupN src/TrainNarrowFOVModel.cpp newsrc/TrainNarrowFOVModel.cpp
--- src/TrainNarrowFOVModel.cpp	2009-07-25 10:57:51.000000000 -0700
+++ newsrc/TrainNarrowFOVModel.cpp	2010-02-10 17:37:00.821137288 -0700
@@ -71,8 +71,7 @@ int main(int argc, char** argv) {
 		cvSetReal2D(subImageGridPoints, i, 0, 15-6*i); //15, 9, 3 
 		cvSetReal2D(subImageGridPoints, i, 1, 15-6*i); //15, 9, 3
 	}
-	char* haarDetectorXMLFile = "data/haarcascade_frontalface_alt2.xml";
-	
+	const char* haarDetectorXMLFile = "data/haarcascade_frontalface_alt2.xml";
 	
 	
 	MIPOMDP* pomdp = new MIPOMDP(inputImageSize, subImageSize, gridSize, 
@@ -93,4 +92,4 @@ int main(int argc, char** argv) {
 	pomdp->saveToFile(savename); 
 	
 	
-}
\ No newline at end of file
+}
