diff -Nurp NMPT_orig/src/BlockTimer.cpp NMPT_ros/src/BlockTimer.cpp
--- NMPT_orig/src/BlockTimer.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/BlockTimer.cpp	2011-04-19 08:39:14.588671042 -0700
@@ -1,4 +1,4 @@
-#include "BlockTimer.h"
+#include <nmpt/BlockTimer.h>
 
 #include <sys/time.h>
 #include <time.h>
diff -Nurp NMPT_orig/src/BoxFeature2.cpp NMPT_ros/src/BoxFeature2.cpp
--- NMPT_orig/src/BoxFeature2.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/BoxFeature2.cpp	2011-04-19 08:39:14.588671042 -0700
@@ -7,8 +7,8 @@
  *
  */
 
-#include "BoxFeature2.h"
-#include "DebugGlobals.h"
+#include "nmpt/BoxFeature2.h"
+#include "nmpt/DebugGlobals.h"
 #include <assert.h>
 #include <stdio.h>
 #include <math.h>
diff -Nurp NMPT_orig/src/BoxFeature.cpp NMPT_ros/src/BoxFeature.cpp
--- NMPT_orig/src/BoxFeature.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/BoxFeature.cpp	2011-04-19 08:39:14.588671042 -0700
@@ -7,8 +7,9 @@
  *
  */
 
-#include "BoxFeature.h"
-#include "DebugGlobals.h"
+#include <nmpt/BoxFeature.h>
+#include <nmpt/DebugGlobals.h>
+#include <cstdio>
 #include <assert.h>
 #include <stdio.h>
 #include <math.h>
diff -Nurp NMPT_orig/src/ChooseSaccadeTargets.cpp NMPT_ros/src/ChooseSaccadeTargets.cpp
--- NMPT_orig/src/ChooseSaccadeTargets.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/ChooseSaccadeTargets.cpp	2011-04-19 08:39:14.588671042 -0700
@@ -17,9 +17,9 @@
 #include <iostream>
 #include <fstream>
 #include <vector>
-#include "BlockTimer.h"
-#include "InternalMotionModel.h"
-#include "NMPTUtils.h"
+#include <nmpt/BlockTimer.h>
+#include <nmpt/InternalMotionModel.h>
+#include <nmpt/NMPTUtils.h>
 
 using namespace cv; 
 using namespace std; 
diff -Nurp NMPT_orig/src/ConvolutionalLogisticPolicy.cpp NMPT_ros/src/ConvolutionalLogisticPolicy.cpp
--- NMPT_orig/src/ConvolutionalLogisticPolicy.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/ConvolutionalLogisticPolicy.cpp	2011-04-19 08:39:14.588671042 -0700
@@ -9,7 +9,7 @@
 
 #include <math.h>
 #include <opencv2/imgproc/imgproc_c.h>
-#include "ConvolutionalLogisticPolicy.h"
+#include <nmpt/ConvolutionalLogisticPolicy.h>
 
 using namespace std; 
 
diff -Nurp NMPT_orig/src/CVPRTestSpeed.cpp NMPT_ros/src/CVPRTestSpeed.cpp
--- NMPT_orig/src/CVPRTestSpeed.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/CVPRTestSpeed.cpp	2011-04-19 08:18:49.047412000 -0700
@@ -47,14 +47,14 @@
 
 
 using namespace std; 
-#include "ImageDataSet.h"
-#include "MIPOMDP.h"
-#include "BlockTimer.h"
+#include "nmpt/ImageDataSet.h"
+#include "nmpt/MIPOMDP.h"
+#include "nmpt/BlockTimer.h"
 #include <stdio.h>
 #include <iostream> 
 #include <string>
 
-#include <opencv2/opencv.hpp>
+#include <opencv2/highgui/highgui_c.h>
 
 
 int main(int argc, char** argv) {
@@ -164,4 +164,4 @@ int main(int argc, char** argv) {
 		delete(test); 
 		delete(pomdp); 
 	}
-}
\ No newline at end of file
+}
diff -Nurp NMPT_orig/src/CVPRTrainModels.cpp NMPT_ros/src/CVPRTrainModels.cpp
--- NMPT_orig/src/CVPRTrainModels.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/CVPRTrainModels.cpp	2011-04-19 08:39:14.588671042 -0700
@@ -46,11 +46,13 @@
  * and Movellan, 2008; Najemnik and Geisler, 2005 (see \ref bib_sec).
  **/
 using namespace std; 
-#include "ImageDataSet.h"
-#include "MIPOMDP.h"
+#include <nmpt/ImageDataSet.h>
+#include <nmpt/MIPOMDP.h>
 #include <stdio.h>
 #include <iostream> 
-#include <opencv2/opencv.hpp>
+
+#include <opencv2/core/core_c.h>
+#include <opencv2/highgui/highgui_c.h>
 
 int main(int argc, char** argv) {
 	const char files[] = "data/GENKI-SZSL_files.txt"; 
diff -Nurp NMPT_orig/src/DetectionEvaluator2.cpp NMPT_ros/src/DetectionEvaluator2.cpp
--- NMPT_orig/src/DetectionEvaluator2.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/DetectionEvaluator2.cpp	2011-04-19 08:39:14.588671042 -0700
@@ -7,10 +7,10 @@
  *
  */
 
-#include "DetectionEvaluator2.h"
+#include "nmpt/DetectionEvaluator2.h"
 #include <iostream>
 #include <sstream>
-#include "NMPTUtils.h"
+#include "nmpt/NMPTUtils.h"
 
 using namespace std;
 using namespace NMPTUtils; 
diff -Nurp NMPT_orig/src/DetectionEvaluator.cpp NMPT_ros/src/DetectionEvaluator.cpp
--- NMPT_orig/src/DetectionEvaluator.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/DetectionEvaluator.cpp	2011-04-19 08:39:14.588671042 -0700
@@ -7,10 +7,10 @@
  *
  */
 
-#include "DetectionEvaluator.h"
+#include <nmpt/DetectionEvaluator.h>
 #include <iostream>
 #include <sstream>
-#include "NMPTUtils.h"
+#include "nmpt/NMPTUtils.h"
 
 using namespace std;
 using namespace NMPTUtils; 
diff -Nurp NMPT_orig/src/EstimateInternalMotion.cpp NMPT_ros/src/EstimateInternalMotion.cpp
--- NMPT_orig/src/EstimateInternalMotion.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/EstimateInternalMotion.cpp	2011-04-19 08:39:14.588671042 -0700
@@ -17,9 +17,9 @@
 #include <iostream>
 #include <fstream>
 
-#include "BlockTimer.h"
-#include "InternalMotionModel.h"
-#include "NMPTUtils.h"
+#include <nmpt/BlockTimer.h>
+#include <nmpt/InternalMotionModel.h>
+#include <nmpt/NMPTUtils.h>
 
 #include <vector>
 #include <string>
diff -Nurp NMPT_orig/src/FastPatchList2.cpp NMPT_ros/src/FastPatchList2.cpp
--- NMPT_orig/src/FastPatchList2.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/FastPatchList2.cpp	2011-04-19 08:39:14.588671042 -0700
@@ -7,8 +7,8 @@
  *
  */
 
-#include "FastPatchList2.h"
-#include "DebugGlobals.h"
+#include "nmpt/FastPatchList2.h"
+#include "nmpt/DebugGlobals.h"
 #include <math.h>
 #include <iostream>
 
diff -Nurp NMPT_orig/src/FastPatchList.cpp NMPT_ros/src/FastPatchList.cpp
--- NMPT_orig/src/FastPatchList.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/FastPatchList.cpp	2011-04-19 08:39:14.588671042 -0700
@@ -7,8 +7,8 @@
  *
  */
 
-#include "FastPatchList.h"
-#include "DebugGlobals.h"
+#include "nmpt/FastPatchList.h"
+#include "nmpt/DebugGlobals.h"
 #include <math.h>
 #include <iostream>
 #include <opencv2/core/core.hpp>
diff -Nurp NMPT_orig/src/FastSalience.cpp NMPT_ros/src/FastSalience.cpp
--- NMPT_orig/src/FastSalience.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/FastSalience.cpp	2011-04-19 08:39:14.588671042 -0700
@@ -1,13 +1,13 @@
-#include "FastSalience.h"
+#include "nmpt/FastSalience.h"
 
 #include <opencv2/imgproc/imgproc.hpp>
 #include <sys/types.h>
 #include <iostream>
 #include <stdio.h>
 #include <stdlib.h>
-#include "NMPTUtils.h" 
-#include "BlockTimer.h"
-#include "DebugGlobals.h"
+#include "nmpt/NMPTUtils.h" 
+#include "nmpt/BlockTimer.h"
+#include "nmpt/DebugGlobals.h"
 
 #define SAL_FILTER_TYPE CV_32F
 
diff -Nurp NMPT_orig/src/FastSUN.cpp NMPT_ros/src/FastSUN.cpp
--- NMPT_orig/src/FastSUN.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/FastSUN.cpp	2011-04-19 08:20:21.874431000 -0700
@@ -10,13 +10,13 @@
 
 
 
-#include <opencv2/opencv.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
 #include <iostream>
 #include <sstream>
-#include "BlockTimer.h"
-#include "FastSalience.h"
-#include "LQRPointTracker.h"
-#include "NMPTUtils.h"
+#include "nmpt/BlockTimer.h"
+#include "nmpt/FastSalience.h"
+#include "nmpt/LQRPointTracker.h"
+#include "nmpt/NMPTUtils.h"
 
 using namespace std;
 using namespace cv; 
@@ -108,4 +108,4 @@ int main (int argc, char * const argv[])
 		
 		imshow("FastSUN Salience", viz); 
 	}
-}
\ No newline at end of file
+}
diff -Nurp NMPT_orig/src/FastSUNImage.cpp NMPT_ros/src/FastSUNImage.cpp
--- NMPT_orig/src/FastSUNImage.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/FastSUNImage.cpp	2011-04-19 08:20:46.843629000 -0700
@@ -10,13 +10,13 @@
 
 
 
-#include <opencv2/opencv.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
 #include <iostream>
 #include <sstream>
-#include "BlockTimer.h"
-#include "FastSalience.h"
-#include "LQRPointTracker.h"
-#include "NMPTUtils.h"
+#include "nmpt/BlockTimer.h"
+#include "nmpt/FastSalience.h"
+#include "nmpt/LQRPointTracker.h"
+#include "nmpt/NMPTUtils.h"
 
 using namespace std;
 using namespace cv; 
@@ -231,4 +231,4 @@ int main (int argc, char * const argv[])
 		
 		imshow(WINDOW_NAME2, viz); 
 	}
-}
\ No newline at end of file
+}
diff -Nurp NMPT_orig/src/Feature2.cpp NMPT_ros/src/Feature2.cpp
--- NMPT_orig/src/Feature2.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/Feature2.cpp	2011-04-19 08:39:14.588671043 -0700
@@ -7,12 +7,12 @@
  *
  */
 
-#include "Feature2.h"
-#include "BoxFeature2.h"
-#include "HaarFeature2.h"
-#include "BlockTimer.h"
-#include "DebugGlobals.h"
-#include "NMPTUtils.h"
+#include "nmpt/Feature2.h"
+#include "nmpt/BoxFeature2.h"
+#include "nmpt/HaarFeature2.h"
+#include "nmpt/BlockTimer.h"
+#include "nmpt/DebugGlobals.h"
+#include "nmpt/NMPTUtils.h"
 
 #include <math.h>
 #include <stdio.h>
diff -Nurp NMPT_orig/src/Feature.cpp NMPT_ros/src/Feature.cpp
--- NMPT_orig/src/Feature.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/Feature.cpp	2011-04-19 08:39:14.588671043 -0700
@@ -7,12 +7,12 @@
  *
  */
 
-#include "Feature.h"
-#include "BoxFeature.h"
-#include "HaarFeature.h"
-#include "BlockTimer.h"
-#include "DebugGlobals.h"
-#include "NMPTUtils.h"
+#include "nmpt/Feature.h"
+#include "nmpt/BoxFeature.h"
+#include "nmpt/HaarFeature.h"
+#include "nmpt/BlockTimer.h"
+#include "nmpt/DebugGlobals.h"
+#include "nmpt/NMPTUtils.h"
 
 #include <math.h>
 #include <stdio.h>
diff -Nurp NMPT_orig/src/FeatureRegressor2.cpp NMPT_ros/src/FeatureRegressor2.cpp
--- NMPT_orig/src/FeatureRegressor2.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/FeatureRegressor2.cpp	2011-04-19 08:39:14.588671043 -0700
@@ -7,11 +7,11 @@
  *
  */
 
-#include "Feature2.h"
-#include "FeatureRegressor2.h"
-#include "DebugGlobals.h"
-#include "NMPTUtils.h"
-#include "ImagePatch.h"
+#include "nmpt/Feature2.h"
+#include "nmpt/FeatureRegressor2.h"
+#include "nmpt/DebugGlobals.h"
+#include "nmpt/NMPTUtils.h"
+#include "nmpt/ImagePatch.h"
 #include <opencv2/core/core.hpp>
 
 using namespace cv;  
diff -Nurp NMPT_orig/src/FeatureRegressor.cpp NMPT_ros/src/FeatureRegressor.cpp
--- NMPT_orig/src/FeatureRegressor.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/FeatureRegressor.cpp	2011-04-19 08:39:14.588671043 -0700
@@ -8,9 +8,9 @@
  */
 
 #include <math.h>
-#include "FeatureRegressor.h"
-#include "DebugGlobals.h"
-#include "NMPTUtils.h"
+#include "nmpt/FeatureRegressor.h"
+#include "nmpt/DebugGlobals.h"
+#include "nmpt/NMPTUtils.h"
 #include <opencv2/core/core.hpp>
 
 using namespace cv;  
diff -Nurp NMPT_orig/src/FoveatedFaceTracker.cpp NMPT_ros/src/FoveatedFaceTracker.cpp
--- NMPT_orig/src/FoveatedFaceTracker.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/FoveatedFaceTracker.cpp	2011-04-19 08:39:14.588671043 -0700
@@ -39,12 +39,15 @@
 
 
 
-#include <opencv2/opencv.hpp>
+#include <opencv2/core/core_c.h>
+#include <opencv2/highgui/highgui_c.h>
+#include <opencv2/imgproc/imgproc_c.h>
+
 #include <cassert>
 #include <iostream>
 #include <fstream>
-#include "MIPOMDP.h"
-#include "BlockTimer.h"
+#include <nmpt/MIPOMDP.h>
+#include <nmpt/BlockTimer.h>
 
 using namespace std;
 
diff -Nurp NMPT_orig/src/FoveatedFaceTrackerImage.cpp NMPT_ros/src/FoveatedFaceTrackerImage.cpp
--- NMPT_orig/src/FoveatedFaceTrackerImage.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/FoveatedFaceTrackerImage.cpp	2011-04-19 08:39:14.588671043 -0700
@@ -40,12 +40,15 @@
 
 
 
-#include <opencv2/opencv.hpp>
+#include <opencv2/core/core_c.h>
+#include <opencv2/highgui/highgui_c.h>
+#include <opencv2/imgproc/imgproc_c.h>
+
 #include <cassert>
 #include <iostream>
 #include <fstream>
-#include "MIPOMDP.h"
-#include "BlockTimer.h"
+#include <nmpt/MIPOMDP.h>
+#include <nmpt/BlockTimer.h>
 
 using namespace std;
 
diff -Nurp NMPT_orig/src/GentleBoostCascadedClassifier.cpp NMPT_ros/src/GentleBoostCascadedClassifier.cpp
--- NMPT_orig/src/GentleBoostCascadedClassifier.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/GentleBoostCascadedClassifier.cpp	2011-04-19 08:39:14.588671043 -0700
@@ -7,13 +7,13 @@
  *
  */
 
-#include "GentleBoostCascadedClassifier.h"
+#include "nmpt/GentleBoostCascadedClassifier.h"
 #include <list>
-#include "BlockTimer.h"
-#include "DebugGlobals.h"
+#include "nmpt/BlockTimer.h"
+#include "nmpt/DebugGlobals.h"
 #include <limits.h>
-#include "HaarFeature.h"
-#include "NMPTUtils.h"
+#include "nmpt/HaarFeature.h"
+#include "nmpt/NMPTUtils.h"
 #include <opencv2/highgui/highgui_c.h>
 #include <opencv2/core/core.hpp>
 #include <opencv2/highgui/highgui.hpp>
diff -Nurp NMPT_orig/src/GentleBoostClassifier2.cpp NMPT_ros/src/GentleBoostClassifier2.cpp
--- NMPT_orig/src/GentleBoostClassifier2.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/GentleBoostClassifier2.cpp	2011-04-19 08:39:14.588671043 -0700
@@ -7,11 +7,11 @@
  *
  */
 
-#include "GentleBoostClassifier2.h"
-#include "Feature2.h"
-#include "DebugGlobals.h" 
-#include "NMPTUtils.h"
-#include "BlockTimer.h"
+#include "nmpt/GentleBoostClassifier2.h"
+#include "nmpt/Feature2.h"
+#include "nmpt/DebugGlobals.h" 
+#include "nmpt/NMPTUtils.h"
+#include "nmpt/BlockTimer.h"
 
 using namespace cv; 
 using namespace std; 
diff -Nurp NMPT_orig/src/GentleBoostClassifier.cpp NMPT_ros/src/GentleBoostClassifier.cpp
--- NMPT_orig/src/GentleBoostClassifier.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/GentleBoostClassifier.cpp	2011-04-19 08:39:14.588671043 -0700
@@ -7,8 +7,8 @@
  *
  */
 
-#include "GentleBoostClassifier.h"
-#include "DebugGlobals.h" 
+#include "nmpt/GentleBoostClassifier.h"
+#include "nmpt/DebugGlobals.h" 
 #include <opencv2/core/core.hpp>
 
 using namespace cv; 
diff -Nurp NMPT_orig/src/HaarFeature2.cpp NMPT_ros/src/HaarFeature2.cpp
--- NMPT_orig/src/HaarFeature2.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/HaarFeature2.cpp	2011-04-19 08:39:14.588671043 -0700
@@ -7,8 +7,8 @@
  *
  */
 
-#include "HaarFeature2.h"
-#include "DebugGlobals.h"
+#include "nmpt/HaarFeature2.h"
+#include "nmpt/DebugGlobals.h"
 #include <math.h>
 #include <iostream>
 #include <stdio.h>
diff -Nurp NMPT_orig/src/HaarFeature.cpp NMPT_ros/src/HaarFeature.cpp
--- NMPT_orig/src/HaarFeature.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/HaarFeature.cpp	2011-04-19 08:39:14.588671043 -0700
@@ -7,8 +7,9 @@
  *
  */
 
-#include "HaarFeature.h"
-#include "DebugGlobals.h"
+#include <nmpt/HaarFeature.h>
+#include <nmpt/DebugGlobals.h>
+#include <cstdio>
 #include <math.h>
 #include <iostream>
 #include <stdio.h>
diff -Nurp NMPT_orig/src/ImageDataSet2.cpp NMPT_ros/src/ImageDataSet2.cpp
--- NMPT_orig/src/ImageDataSet2.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/ImageDataSet2.cpp	2011-04-19 08:39:14.588671043 -0700
@@ -7,7 +7,7 @@
  *
  */
 
-#include "ImageDataSet2.h"
+#include "nmpt/ImageDataSet2.h"
 #include <fstream>
 #include <stdlib.h>
 #include <string.h>
diff -Nurp NMPT_orig/src/ImageDataSet2OpenCVDescriptionFormat.cpp NMPT_ros/src/ImageDataSet2OpenCVDescriptionFormat.cpp
--- NMPT_orig/src/ImageDataSet2OpenCVDescriptionFormat.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/ImageDataSet2OpenCVDescriptionFormat.cpp	2011-04-19 08:39:14.588671043 -0700
@@ -9,12 +9,13 @@
 
 
 
-#include <opencv2/opencv.hpp>
+#include <opencv2/core/core_c.h>
+#include <opencv2/highgui/highgui_c.h>
 #include <string>
 #include <iostream>
 #include <fstream>
-#include "DetectionEvaluator.h"
-#include "ImageDataSet.h"
+#include <nmpt/DetectionEvaluator.h>
+#include <nmpt/ImageDataSet.h>
 
 using namespace std; 
 
diff -Nurp NMPT_orig/src/ImageDataSet.cpp NMPT_ros/src/ImageDataSet.cpp
--- NMPT_orig/src/ImageDataSet.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/ImageDataSet.cpp	2011-04-19 08:39:14.588671043 -0700
@@ -7,7 +7,7 @@
  *
  */
 
-#include "ImageDataSet.h"
+#include <nmpt/ImageDataSet.h>
 #include <fstream>
 #include <stdlib.h>
 #include <string.h>
diff -Nurp NMPT_orig/src/ImageKalmanFilter.cpp NMPT_ros/src/ImageKalmanFilter.cpp
--- NMPT_orig/src/ImageKalmanFilter.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/ImageKalmanFilter.cpp	2011-04-19 08:39:14.588671043 -0700
@@ -7,7 +7,7 @@
  *
  */
 
-#include "ImageKalmanFilter.h"
+#include <nmpt/ImageKalmanFilter.h>
 #include <opencv2/imgproc/imgproc_c.h>
 
 using namespace cv; 
diff -Nurp NMPT_orig/src/ImagePatch2.cpp NMPT_ros/src/ImagePatch2.cpp
--- NMPT_orig/src/ImagePatch2.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/ImagePatch2.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,9 +7,9 @@
  *
  */
 
-#include "ImagePatch2.h"
-#include "DebugGlobals.h"
-#include "NMPTUtils.h"
+#include "nmpt/ImagePatch2.h"
+#include "nmpt/DebugGlobals.h"
+#include "nmpt/NMPTUtils.h"
 #include <iostream>
 #include <math.h>
 #include <fstream>
diff -Nurp NMPT_orig/src/ImagePatch.cpp NMPT_ros/src/ImagePatch.cpp
--- NMPT_orig/src/ImagePatch.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/ImagePatch.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,9 +7,9 @@
  *
  */
 
-#include "ImagePatch.h"
-#include "FeatureRegressor.h"
-#include "BoxFeature.h"
+#include <nmpt/ImagePatch.h>
+#include <nmpt/FeatureRegressor.h>
+#include <nmpt/BoxFeature.h>
 #include <iostream>
 #include <math.h>
 #include <fstream>
diff -Nurp NMPT_orig/src/ImagePatchPyramid.cpp NMPT_ros/src/ImagePatchPyramid.cpp
--- NMPT_orig/src/ImagePatchPyramid.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/ImagePatchPyramid.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,8 +7,8 @@
  *
  */
 
-#include "ImagePatchPyramid.h"
-#include "DebugGlobals.h"
+#include <nmpt/ImagePatchPyramid.h>
+#include <nmpt/DebugGlobals.h>
 #include <fstream>
 #include <stdio.h>
 #include <opencv2/imgproc/imgproc_c.h>
diff -Nurp NMPT_orig/src/InternalMotionModel.cpp NMPT_ros/src/InternalMotionModel.cpp
--- NMPT_orig/src/InternalMotionModel.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/InternalMotionModel.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,10 +7,10 @@
  *
  */
 
-#include "InternalMotionModel.h"
-#include "MatrixKalmanFilter.h"
-#include "ImageKalmanFilter.h"
-#include "NMPTUtils.h"
+#include <nmpt/InternalMotionModel.h>
+#include <nmpt/MatrixKalmanFilter.h>
+#include <nmpt/ImageKalmanFilter.h>
+#include <nmpt/NMPTUtils.h>
 
 using namespace NMPTUtils; 
 using namespace std; 
diff -Nurp NMPT_orig/src/KF.cpp NMPT_ros/src/KF.cpp
--- NMPT_orig/src/KF.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/KF.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -6,7 +6,7 @@
 //  Copyright (c) 2010 __MyCompanyName__. All rights reserved.
 //
 
-#include "KF.h"
+#include <nmpt/KF.h>
 
 using namespace cv; 
 
diff -Nurp NMPT_orig/src/KFPointTracker.cpp NMPT_ros/src/KFPointTracker.cpp
--- NMPT_orig/src/KFPointTracker.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/KFPointTracker.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,8 +7,8 @@
  *
  */
 
-#include "KFPointTracker.h"
-#include "KF.h"
+#include <nmpt/KFPointTracker.h>
+#include <nmpt/KF.h>
 
 using namespace cv; 
 
diff -Nurp NMPT_orig/src/LQR.cpp NMPT_ros/src/LQR.cpp
--- NMPT_orig/src/LQR.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/LQR.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -6,7 +6,7 @@
 //  Copyright (c) 2010 __MyCompanyName__. All rights reserved.
 //
 
-#include "LQR.h"
+#include <nmpt/LQR.h>
 
 using namespace cv; 
 
diff -Nurp NMPT_orig/src/LQRPointTracker.cpp NMPT_ros/src/LQRPointTracker.cpp
--- NMPT_orig/src/LQRPointTracker.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/LQRPointTracker.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,7 +7,7 @@
  *
  */
 
-#include "LQRPointTracker.h"
+#include <nmpt/LQRPointTracker.h>
 
 using namespace cv; 
 
diff -Nurp NMPT_orig/src/MatrixKalmanFilter.cpp NMPT_ros/src/MatrixKalmanFilter.cpp
--- NMPT_orig/src/MatrixKalmanFilter.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/MatrixKalmanFilter.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,7 +7,7 @@
  *
  */
 
-#include "MatrixKalmanFilter.h"
+#include <nmpt/MatrixKalmanFilter.h>
 
 using namespace cv; 
 using namespace std; 
diff -Nurp NMPT_orig/src/MIPOMDP.cpp NMPT_ros/src/MIPOMDP.cpp
--- NMPT_orig/src/MIPOMDP.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/MIPOMDP.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,8 +7,8 @@
  *
  */
 
-#include "MIPOMDP.h"
-#include "DebugGlobals.h"
+#include <nmpt/MIPOMDP.h>
+#include <nmpt/DebugGlobals.h>
 #include <math.h>
 #include <stdlib.h>
 #include <cassert>
diff -Nurp NMPT_orig/src/MultinomialObservationModel.cpp NMPT_ros/src/MultinomialObservationModel.cpp
--- NMPT_orig/src/MultinomialObservationModel.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/MultinomialObservationModel.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,7 +7,7 @@
  *
  */
 
-#include "MultinomialObservationModel.h"
+#include <nmpt/MultinomialObservationModel.h>
 
 using namespace std; 
 
diff -Nurp NMPT_orig/src/MultiObjectTrackingState.cpp NMPT_ros/src/MultiObjectTrackingState.cpp
--- NMPT_orig/src/MultiObjectTrackingState.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/MultiObjectTrackingState.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,8 +7,8 @@
  *
  */
 
-#include "MultiObjectTrackingState.h"
-#include "DebugGlobals.h"
+#include <nmpt/MultiObjectTrackingState.h>
+#include <nmpt/DebugGlobals.h>
 #include <iostream>
 
 using namespace std; 
diff -Nurp NMPT_orig/src/NMPTUtils.cpp NMPT_ros/src/NMPTUtils.cpp
--- NMPT_orig/src/NMPTUtils.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/NMPTUtils.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,8 +7,8 @@
  *
  */
 
-#include "NMPTUtils.h"
-#include "DebugGlobals.h"
+#include "nmpt/NMPTUtils.h"
+#include "nmpt/DebugGlobals.h"
 #include <iostream>
 #include <sstream>
 #include <opencv2/imgproc/imgproc.hpp>
diff -Nurp NMPT_orig/src/ObjectDetector.cpp NMPT_ros/src/ObjectDetector.cpp
--- NMPT_orig/src/ObjectDetector.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/ObjectDetector.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,7 +7,7 @@
  *
  */
 
-#include "ObjectDetector.h"
+#include <nmpt/ObjectDetector.h>
 using namespace std; 
 
 ObjectDetector::~ObjectDetector(){}
diff -Nurp NMPT_orig/src/OpenCV2BoxFilter.cpp NMPT_ros/src/OpenCV2BoxFilter.cpp
--- NMPT_orig/src/OpenCV2BoxFilter.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/OpenCV2BoxFilter.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -9,7 +9,7 @@
 
 #include <opencv2/imgproc/imgproc.hpp>
 #include <iostream>
-#include "OpenCV2BoxFilter.h"
+#include "nmpt/OpenCV2BoxFilter.h"
 
 using namespace std; 
 using namespace cv; 
diff -Nurp NMPT_orig/src/OpenCVHaarDetector.cpp NMPT_ros/src/OpenCVHaarDetector.cpp
--- NMPT_orig/src/OpenCVHaarDetector.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/OpenCVHaarDetector.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,7 +7,7 @@
  *
  */
 
-#include "OpenCVHaarDetector.h"
+#include <nmpt/OpenCVHaarDetector.h>
 
 using namespace cv; 
 using namespace std; 
diff -Nurp NMPT_orig/src/OpenLoopPolicies.cpp NMPT_ros/src/OpenLoopPolicies.cpp
--- NMPT_orig/src/OpenLoopPolicies.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/OpenLoopPolicies.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -8,7 +8,7 @@
  */
 
 #include <iostream>
-#include "OpenLoopPolicies.h"
+#include <nmpt/OpenLoopPolicies.h>
 
 using namespace std; 
 
diff -Nurp NMPT_orig/src/PatchDataset2.cpp NMPT_ros/src/PatchDataset2.cpp
--- NMPT_orig/src/PatchDataset2.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/PatchDataset2.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,16 +7,16 @@
  *
  */
 
-#include "PatchDataset2.h"
+#include "nmpt/PatchDataset2.h"
 #include <math.h>
 #include <fstream>
 #include <opencv2/imgproc/imgproc.hpp>
 #include <opencv2/highgui/highgui.hpp>
-#include "DebugGlobals.h"
-#include "DetectionEvaluator.h"
-#include "OpenLoopPolicies.h"
-#include "FastPatchList.h"
-#include "NMPTUtils.h"
+#include "nmpt/DebugGlobals.h"
+#include "nmpt/DetectionEvaluator.h"
+#include "nmpt/OpenLoopPolicies.h"
+#include "nmpt/FastPatchList.h"
+#include "nmpt/NMPTUtils.h"
 
 using namespace std; 
 using namespace cv;
diff -Nurp NMPT_orig/src/PatchDataset.cpp NMPT_ros/src/PatchDataset.cpp
--- NMPT_orig/src/PatchDataset.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/PatchDataset.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -9,17 +9,17 @@
 
 
 
-#include "PatchDataset.h"
+#include "nmpt/PatchDataset.h"
 
 #include <math.h>
 #include <fstream>
 #include <opencv2/imgproc/imgproc.hpp>
 #include <opencv2/imgproc/imgproc_c.h>
 #include <opencv2/highgui/highgui_c.h>
-#include "DetectionEvaluator.h"
-#include "OpenLoopPolicies.h"
-#include "FastPatchList.h"
-#include "NMPTUtils.h"
+#include "nmpt/DetectionEvaluator.h"
+#include "nmpt/OpenLoopPolicies.h"
+#include "nmpt/FastPatchList.h"
+#include "nmpt/NMPTUtils.h"
 
 using namespace std; 
 using namespace cv;
diff -Nurp NMPT_orig/src/PatchList2.cpp NMPT_ros/src/PatchList2.cpp
--- NMPT_orig/src/PatchList2.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/PatchList2.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,8 +7,8 @@
  *
  */
 
-#include "PatchList2.h"
-#include "DebugGlobals.h"
+#include "nmpt/PatchList2.h"
+#include "nmpt/DebugGlobals.h"
 #include <math.h>
 #include <iostream>
 #include <stdlib.h>
diff -Nurp NMPT_orig/src/PatchList.cpp NMPT_ros/src/PatchList.cpp
--- NMPT_orig/src/PatchList.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/PatchList.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -7,8 +7,8 @@
  *
  */
 
-#include "PatchList.h"
-#include "DebugGlobals.h"
+#include <nmpt/PatchList.h>
+#include <nmpt/DebugGlobals.h>
 #include <math.h>
 #include <iostream>
 #include <stdlib.h>
diff -Nurp NMPT_orig/src/SimpleFaceTracker.cpp NMPT_ros/src/SimpleFaceTracker.cpp
--- NMPT_orig/src/SimpleFaceTracker.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/SimpleFaceTracker.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -72,8 +72,8 @@
 #include <cassert>
 #include <iostream>
 #include <fstream>
-#include "MIPOMDP.h"
-#include "BlockTimer.h"
+#include "nmpt/MIPOMDP.h"
+#include "nmpt/BlockTimer.h"
 
 using namespace std;
 
diff -Nurp NMPT_orig/src/SimpleSalienceExample.cpp NMPT_ros/src/SimpleSalienceExample.cpp
--- NMPT_orig/src/SimpleSalienceExample.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/SimpleSalienceExample.cpp	2011-04-19 08:21:54.451457000 -0700
@@ -19,11 +19,12 @@
 
 
 
-#include <opencv2/opencv.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/highgui/highgui.hpp>
 #include <vector>
 #include <string>
 #include <iostream>
-#include "FastSalience.h"
+#include "nmpt/FastSalience.h"
 
 using namespace std;
 using namespace cv; 
@@ -75,4 +76,4 @@ int main (int argc, char * const argv[])
 		
 		
 	} 
-}
\ No newline at end of file
+}
diff -Nurp NMPT_orig/src/StructTypes.cpp NMPT_ros/src/StructTypes.cpp
--- NMPT_orig/src/StructTypes.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/StructTypes.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -1,5 +1,5 @@
-#include "StructTypes.h"
-#include "GentleBoostClassifier2.h"
+#include "nmpt/StructTypes.h"
+#include "nmpt/GentleBoostClassifier2.h"
 
 double (*_featureCost)(const PerformanceMetrics& ) = &(GentleBoostClassifier2::featureCost);
 
diff -Nurp NMPT_orig/src/TrainNarrowFOVModel.cpp NMPT_ros/src/TrainNarrowFOVModel.cpp
--- NMPT_orig/src/TrainNarrowFOVModel.cpp	2010-12-09 17:05:54.000000000 -0700
+++ NMPT_ros/src/TrainNarrowFOVModel.cpp	2011-04-19 08:39:14.598670730 -0700
@@ -47,11 +47,12 @@
  * and Movellan, 2008; Najemnik and Geisler, 2005 (see \ref bib_sec).
  **/
 using namespace std; 
-#include "ImageDataSet.h"
-#include "MIPOMDP.h"
+#include <nmpt/ImageDataSet.h>
+#include <nmpt/MIPOMDP.h>
 #include <stdio.h>
 #include <iostream> 
-#include <opencv2/opencv.hpp>
+#include <opencv2/core/core_c.h>
+#include <opencv2/highgui/highgui_c.h>
 
 int main(int argc, char** argv) {
 	const char files[] = "data/GENKI-SZSL_files.txt"; 
--- NMPT_orig/src/PatchList.cpp	2012-02-05 01:46:16.979385902 +0900
+++ NMPT_ros/src/PatchList.cpp	2012-02-05 01:47:11.967383956 +0900
@@ -461,7 +461,7 @@
 		scaleSize = getEffectiveResampledImageSizeAtScale(i); 
 		
 		Mat resizeScratchPad = scaleCanvas(Rect(0, 0, scaleSize.width, scaleSize.height)); 
-		resize(newImage, resizeScratchPad, resizeScratchPad.size(), 0,0, INTER_NEAREST); 
+		resize(Mat(newImage), resizeScratchPad, resizeScratchPad.size(), 0,0, INTER_NEAREST); 
 		
 		if (_PATCHLIST_DEBUG) cout << "Resized image." << endl; 
 		if ((unsigned int) i < images.size()) {	
--- NMPT_orig/src/NMPTUtils.cpp.org	2012-02-05 01:46:27.471385531 +0900
+++ NMPT_ros/src/NMPTUtils.cpp	2012-02-05 01:46:36.535385211 +0900
@@ -149,7 +149,7 @@
 	
 	int useFast = numels > 10; 
 	if (useFast) {
-		sort(randEls, randEls, CV_SORT_EVERY_ROW | CV_SORT_ASCENDING);
+		cv::sort(randEls, randEls, CV_SORT_EVERY_ROW | CV_SORT_ASCENDING);
 		int h = 0; 
 		double c = normHist.at<double>(0,h); 
 		for (int i = 0; i < numels; i++) {
